src/internet_identity_interface/src/lib.rs

```diff
@@ -173,3 +173,92 @@ pub struct HttpResponse {
 pub struct InternetIdentityInit {
     pub assigned_user_number_range: (UserNumber, UserNumber),
 }
+
+// Archive specific types
+
+#[derive(Eq, PartialEq, Clone, Debug, CandidType, Deserialize)]
+pub enum OperationType {
```

Comment:
throughout this type it may be nice to not repeat the operation name in each `..._device` field, but instead simply use `device: ...`, which may also be much nicer to use for any client written in a language with row polymorphism